profiles/common/modules/features/nexteuropa_dgt_connector/tmgmt_dgt_connector/src/Subscriber.php

```diff
@@ -0,0 +1,372 @@
+<?php
+
+/**
+ * @file
+ * Provides events subscriber for TMGMT DGT Connector.
+ */
+
+namespace Drupal\tmgmt_dgt_connector;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use EC\Poetry\Events\Notifications\TranslationReceivedEvent;
+use EC\Poetry\Events\Notifications\StatusUpdatedEvent;
+
+/**
+ * Subscriber with listeners for Server events.
+ *
+ * @package Drupal\tmgmt_dgt_connector
+ */
+class Subscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return array(
+      TranslationReceivedEvent::NAME => 'onTranslationReceivedEvent',
+      StatusUpdatedEvent::NAME => 'onStatusUpdatedEvent',
+    );
+  }
+
+  /**
+   * Listener for the event onTranslationReceivedEvent.
+   *
+   * @param \EC\Poetry\Events\Notifications\TranslationReceivedEvent $event
+   *   The event for the Translation Received.
+   *
+   * @return bool
+   *   Return True if the translation is received without issues.
+   */
+  public function onTranslationReceivedEvent(TranslationReceivedEvent $event) {
+    $translator = tmgmt_translator_load(TMGMT_DGT_CONNECTOR_TRANSLATOR_NAME);
```

Comment:
I think both onTranslationReceivedEvent and onStatusUpdatedEvent should check if the translator associated to a job is the tmgmt_dgt_connector one or not. This will prevent it from triggering on other translators (like Fast Track)

Reply:
You're right @imanoleguskiza . This is the same situation we faced in Fast Track.
But I propose to postpone that change to a new ticket because of two strong reasons:
- This ticket/PR has consumed already a huge amount of time and, as pledged in the retrospective meeting, we should create more tickets with precise scope (this is the same approach we had for Fast Track, where a special ticket was created, [NEPT-1590](https://webgate.ec.europa.eu/CITnet/jira/browse/NEPT-1590));
- Within [NEPT-1601](https://webgate.ec.europa.eu/CITnet/jira/browse/NEPT-1601) we are planning to handle this logic in a proper way, so that we don't need to do the same steps (check reference, get job, get translator, then proceed or skip the request) in each listener.